abz/acousticbrainz.py

```diff
@@ -164,3 +164,185 @@ def process(path):
+    datasetdict = _datasetdict_structure(dataset_name)
+    itemuuid = None
+    
+    for dirName, sudDirNames, fileNames in os.walk(path):
+        
+        for subdir in sudDirNames:
+            tmpdict = {}
+            tmpdict['name'] = subdir
+            tmpdict['description'] = ''
+            tmpdict['recordings'] = []
+
+            # scan each subfolder in the dataset root
+            subdir = os.path.join(path, subdir)
+    
+            for dirName, subDirNames, fileNames in os.walk(subdir):
+                ### run the extractor
+                for fileName in fileNames:
+                    # run the extractor
+                    filepath = os.path.join(subdir, fileName)
+                    extractor_output = _dataset_item_extractor(filepath)
+                    # submit dataset item (md5 check)
+                    itemuuid = submit_dataset_item(extractor_output)
+                    if not itemuuid==None:
+                        itemuuid = itemuuid['itemuuid']
+                        tmpdict['recordings'].append(str(itemuuid))
+                        _update_progress(filepath, ":) Get UUID", GREEN)
+                        print()
+                    
+                    os.unlink(extractor_output)
+                    
+                if len(tmpdict['recordings'])>0:
+                    datasetdict['classes'].append(tmpdict)
+    
+    #print(datasetdict)
+    return datasetdict
+
+def _dataset_item_extractor(filepath):
+    """Extract lowlevel data from a dataset item
+    
+    Args:
+        filepath:   the fill path of the dataset item to process
+    Returns:
+        tmpname:    the full path of the generated lowlevel-data json file
+    """
+    fd, tmpname = tempfile.mkstemp(suffix='.json')
+    os.close(fd)
+    os.unlink(tmpname)
+    # avoid using default config file for datasets items MBID is not required
+    config.settings["profile_file"] = ""
+    try:
+        retcode, out = run_extractor(filepath, tmpname)
+        if retcode == 1:
+            print()
+            print(out)            
+            _update_progress(filepath, ":( extract", RED)
+            add_to_filelist(filepath, "extractor")
+        elif retcode < 0 or retcode > 0:
+            print()
+            print(out)
+            _update_progress(filepath, ":( unknown error", RED)
+        else:
+            if os.path.isfile(tmpname):
+                return tmpname
+            else:
+                return None
+    except KeyboardInterrupt:
+        print()
+        print("Action interrupted by the user!")
+        print("Stopped while extracting data from file:", filepath)
+        sys.exit(1)
+
+def _datasetdict_structure(dataset_name):
+    """Generates the skeleton for the dataset dictionary
+    
+    Args:
+        dataset_name:   the name of the dataset (the root folder of the dataset)
+    Returns:
+        datasetdict:    a dictionar ywith the dataset structure
+    """
+    datasetdict = {}
```

Comment:
This would be clearer if you made it as a dictionary literal:
```
datasetdict = {"name": dataset_name,
                      "description": "",
    ...etc
```